ci-automation/tapfile_helper_lib.sh

```diff
@@ -0,0 +1,281 @@
+        sqlite3 "${dbfile}" "$@"
+        if [ $? -ne 5 ] ; then
+            return $?
+        fi
+        local sleep="$((1 + $RANDOM % 5))"
+        echo "Retrying in ${sleep} seconds." >&2
+        sleep "${sleep}"
+    done
+}
+# --
+
+# Initialise the DB if it wasn't yet.
+function __db_init() {
+    local dbname="${TAPFILE_HELPER_DBNAME}"
+
+    __sqlite3_wrapper "${dbname}" '
+    CREATE TABLE IF NOT EXISTS "test_case" (
+        "id"    INTEGER,
+        "name"  TEXT UNIQUE,
+        PRIMARY KEY("id")
+    );
+    CREATE TABLE IF NOT EXISTS "vendor" (
+        "id"    INTEGER,
+        "name"  TEXT UNIQUE,
+        PRIMARY KEY("id")
+    );
+    CREATE TABLE IF NOT EXISTS "test_run" (
+        "id"        INTEGER NOT NULL,
+        "result"    INTEGER NOT NULL,
+        "output"    TEXT,
+        "case_id"   INTEGER NOT NULL,
+        "run"       INTEGER NOT NULL,
+        "vendor_id" INTEGER,
+        PRIMARY KEY("id"),
+        FOREIGN KEY("case_id") REFERENCES "test_case"("id"),
+        FOREIGN KEY("vendor_id") REFERENCES "vendor"("id"),
+        UNIQUE (case_id, run, vendor_id)
+    );
+'
+}
+# --
+
+# Read tapfile into temporary DB.
+# INPUT:
+# 1: <tapfile> - tapfile to ingest
+# 2: <vendor>  - vendor (qemu, azure, aws, etc...)
+# 3: <run>     - re-run iteration
+
+function tap_ingest_tapfile() {
+    local tapfile="${1}"
+    local vendor="${2}"
+    local run="${3}"
+
+    local dbname="${TAPFILE_HELPER_DBNAME}"
+
+    local result=""
+    local test_name=""
+    local error_message=""
+    local in_error_message=false
+
+    if ! [ -f "${TAPFILE_HELPER_DBNAME}" ] ; then
+       __db_init
+    fi
+
+    # Wrap all SQL commands in a transaction to speed up INSERTs
+    local SQL="BEGIN TRANSACTION;"
+
+    # Example TAP input:
+    # ok - coreos.auth.verify
+    # ok - coreos.locksmith.tls
+    # not ok - cl.filesystem
+    #   ---
+    #   Error: "--- FAIL: cl.filesystem/deadlinks (1.86s)\n            files.go:90: Dead symbolic links found: [/var/lib/flatcar-oem-gce/usr/lib64/python3.9/site-packages/certifi-3021.3.16-py3.9.egg-info]"
+    #   ...
+    # ok - cl.cloudinit.script
+    # ok - kubeadm.v1.22.0.flannel.base
+    while read -r line; do
+        if [[ "${line}" == "1.."* ]] ; then continue; fi
+        if [ "${line}" = "---" ] ; then  # note: read removes leading whitespaces
+            in_error_message=true
+            continue
+        fi
+
+        if $in_error_message ; then
+            if [ "${line}" = "..." ] ; then
+                in_error_message=false
+            else
+                error_message="$(echo -e "$line" \
+                                    | sed -e 's/^Error: "--- FAIL: /"/' -e 's/^[[:space:]]*//' \
+                                    | sed -e "s/[>\"']/_/g" -e 's/[[:space:]]/ /g')"
+                continue
+            fi
+        else
+            test_name="$(echo "${line}" | sed 's/^[^-]* - //')"
+            local result_string="$(echo "${line}" | sed 's/ - .*//')"
```

Comment:
```suggestion
            local result_string
            result_string="$(echo "${line}" | sed 's/ - .*//')"
```
otherwise the error code of the $() evaluation is masked